HW9/untitled/src/Manager.java

```diff
@@ -0,0 +1,39 @@
+public class Manager extends Employee {
+    Manager(String name, int age, double salary) {
+        super(name, age, salary);
+    }
+
+    public void work(){
+        System.out.println("I'm working as a MANAGER. I have a lot of duties");
+        System.out.println("My name is "+ getName());
+        System.out.println("My salary is "+ getSalary());
+    }
+
+    public void takeBreak(){
+        System.out.println("I am out, at lunch as a MANAGER");
+    }
+
+    public void performJob(){
+        System.out.println("I performing a job as a MANAGER. I have to control everything");
+    }
+
+
+    public void describeRole(){
+        System.out.println("My role is manager. It's my repsonsibility to make sure we/ll finish in time");
+
+    }
+
+    public void describeRole(String extraInfo, int subordinates){
```

Comment:
The overall idea looks good, although if `subordinates` is not a property of the manager, better we don't mention it as a parameter as it's not a part of the objects' state.

According to this implementation, whoever calls the method can 'tell' how many subordinates manager has and manager itself takes it for granted and displays it despite it doesn't know anything about its subordinates

OR

add a subordinates field (array or just an int) and make method .addSubordinate() so manager will hold his/her subordinates as a part of the object state